### DISCUSSION TITLE
posts/sample_microfacet_brdf/

### DISCUSSION THREAD

**Discussion by giscus** [JiayinCao/AGraphicsGuyNotes#15] [Giscus-Comments]
# posts/sample_microfacet_brdf/

I’m working on microfacet brdf model for my renderer these days, noticing that it is more than necessary to provide a separate sampling method for microfacet brdf instead of using the default one, which is usually used for diffuse like surfaces and highly inefficient for brdf with spiky shape, such as mirror like surfaces. The following image is one generated by the default sampling method:
The left monkey has pure reflection brdf which is mentioned in my previous blog post, the right one uses the microfacet model with zero as roughness value.

https://agraphicsguynotes.com/posts/sample_microfacet_brdf/

**Comment by Kaminate** [comment]
Reference link [4] has moved to https://www.tobias-franke.eu/log/2014/03/30/notes_on_importance_sampling.html

**Reply by JiayinCao** [reply]
It is a bit hard to make sure all the referenece links are still valid in my blog posts since some of them are pretty old. People's blog could migrate to other places, easily making links invalid.

W.r.t this one, it is already updated. Thanks for letting me know.

**Comment by cecerawrxd** [comment]
Can you please explain how you derived the formula for the pdf of the incidence ray from the pdf of the randomly sampled normal? I would greatly appreciate it.

**Reply by JiayinCao** [reply]
Please take a look at Pbrt for further details.
[Here](https://pbr-book.org/3ed-2018/Light_Transport_I_Surface_Reflection/Sampling_Reflection_Functions#MicrofacetBxDFs)  is a link to the pbrt chapter that explains it.

**Reply by cecerawrxd** [reply]
I see. Thanks for replying.

**Comment by neverhood311** [comment]
In the derivations for the CDFs for each of the three normal distribution functions, you start by integrating from 0 to $\theta$ but then flip to $\theta$ to 0. Why is that?
